src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -814,4 +723,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  override def infosByExecution(id: WorkflowId): Future[Traversable[ExecutionInfosByExecution]] = {
+    val action = for {
+      rawTuples <- dataAccess.executionsAndExecutionInfosByWorkflowId(id.toString).result
+      groupedTuples = rawTuples groupBy { _._1 } mapValues { _ map { _._2 } }
```

Comment:
I think your original comment about replacing "tuples" with a real type will go a long way to making this more readable